internal/controller/pkg/revision/deployment.go

```diff
@@ -75,8 +80,14 @@ func buildProviderDeployment(provider *metav1alpha1.Provider, revision v1beta1.P
 					ImagePullSecrets:   revision.GetPackagePullSecrets(),
 					Containers: []corev1.Container{
 						{
-							Name:            provider.GetName(),
-							Image:           provider.Spec.Controller.Image,
+							Name:  provider.GetName(),
+							Image: provider.Spec.Controller.Image,
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "metrics",
+									ContainerPort: 8080,
+								},
+							},
```

Comment:
@khos2ow I would prefer to let this be customized by usage of the [ControllerConfig](https://doc.crds.dev/github.com/crossplane/crossplane/pkg.crossplane.io/ControllerConfig/v1alpha1) as there is nothing that dictates that a provider expose metrics in this manner

Reply:
Originally I thought about extending `ControllerConfig` to pass port and annotations, but ultimately providers are built on top of `controller-runtime` which exposes metric server anyways. So if a provider's deployment exposes that port we're effectively in business of gathering the metrics. And I  figured if the Crossplane admin has the intention of "enabling metrics", centrally from core chart, it might be nice to propagate that intention to providers without any extra actions need to be taken.

On the other hand I think your argument still stands for the above annotations which auto scrapes metrics, and that might be a good candidate to move to ControllerConfig, IMO.

Reply:
Yeah the issue is that nothing says provider _have_ to use `controller-runtime`. The annotations already are passed through, we could consider adding the ports as well

Reply:
Oh I see, fair point. I'm gonna revert this and squash the original commits and this should be good to go then. I'll open another PR for port as well, just in case.

Reply:
Awesome! Thanks again for your work here 🙌

Reply:
For sure, happy to be here!